extensions/standard-processors/processors/ListenSyslog.cpp

```diff
@@ -17,318 +14,272 @@
+const std::regex ListenSyslog::SyslogMessage::rfc3164_pattern_(
+    R"((?:\<(\d{1,3})\>))"                                                                                    // priority
+    R"(([A-Z][a-z][a-z]\s{1,2}\d{1,2}\s\d{2}[:]\d{2}[:]\d{2})\s)"                                             // timestamp
+    R"(([\w][\w\d(\.|\:)@-]*)\s)"                                                                             // hostname
+    R"((.*)$)", std::regex::ECMAScript);                                                                      // msg
 
 void ListenSyslog::initialize() {
-  // Set the supported properties
-  std::set<core::Property> properties;
-  properties.insert(RecvBufSize);
-  properties.insert(MaxSocketBufSize);
-  properties.insert(MaxConnections);
-  properties.insert(MaxBatchSize);
-  properties.insert(MessageDelimiter);
-  properties.insert(ParseMessages);
-  properties.insert(Protocol);
-  properties.insert(Port);
-  setSupportedProperties(properties);
-  // Set the supported relationships
-  std::set<core::Relationship> relationships;
-  relationships.insert(Success);
-  relationships.insert(Invalid);
-  setSupportedRelationships(relationships);
+  setSupportedProperties({Port, ProtocolProperty, MaxBatchSize, ParseMessages, MaxQueueSize});
+  setSupportedRelationships({Success, Invalid});
 }
 
-void ListenSyslog::startSocketThread() {
-  if (_thread != NULL)
-    return;
+void ListenSyslog::onSchedule(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::ProcessSessionFactory>&) {
+  gsl_Expects(context && !server_thread_.joinable() && !server_);
 
-  logger_->log_trace("ListenSysLog Socket Thread Start");
-  _serverTheadRunning = true;
-  _thread = new std::thread(run, this);
-  _thread->detach();
-}
+  context->getProperty(MaxBatchSize.getName(), max_batch_size_);
+  context->getProperty(ParseMessages.getName(), parse_messages_);
 
-void ListenSyslog::run(ListenSyslog *process) {
-  process->runThread();
-}
+  uint64_t max_queue_size = 0;
+  context->getProperty(MaxQueueSize.getName(), max_queue_size);
+  max_queue_size_ = max_queue_size > 0 ? std::optional<uint64_t>(max_queue_size) : std::nullopt;
 
-void ListenSyslog::runThread() {
-  while (_serverTheadRunning) {
-    if (_resetServerSocket) {
-      _resetServerSocket = false;
-      // need to reset the socket
-      std::vector<int>::iterator it;
-      for (it = _clientSockets.begin(); it != _clientSockets.end(); ++it) {
-        int clientSocket = *it;
-        close(clientSocket);
-      }
-      _clientSockets.clear();
-      if (_serverSocket > 0) {
-        close(_serverSocket);
-        _serverSocket = 0;
-      }
-    }
+  Protocol protocol = Protocol(ProtocolProperty.getDefaultValue());
+  if (!context->getProperty(ProtocolProperty.getName(), protocol)) {
+    logger_->log_error("Missing or invalid Protocol: defaulting to %s", protocol.toString());
```

Comment:
Not sure if this can occur anyhow given the validator, but in case it does, it should throw

Reply:
Good point. I checked and since these are required properties the context->getProperty will already throw if it can't find the property.
So these branches are unreachable. I removed them in https://github.com/apache/nifi-minifi-cpp/pull/1294/commits/251865e006f1e53e45c33fdaa184a9259f7bf13d#diff-a22ae2c37869a184bd936ac719f98ae8bbdb1b8d2ad5b3e6fc48ea9207fe530bR95-R99